src/components/ProductItem/ProductItem.jsx

```diff
@@ -1,21 +1,61 @@
+import { useAuth, useWishlist } from '../../contexts';
+import { useCart } from '../../contexts/cart-context';
+
 import './product-item.css';
 
 const ProductItem = ({ product }) => {
+    const { auth } = useAuth();
+    const { cartItems, addToCartHandler, updateHandler } = useCart();
+    const { wishlistItems, addToWishlistHandler, removeFromWishlistHandler } = useWishlist();
+    const navigate = useNavigate();
+
+    const { _id, asset, title, price, discount } = product;
+
+    const checkWishlist = wishlistItems.items.find((item) => item._id === _id);
+    const checkCart = cartItems.items.find((item) => item._id === _id);
+
     return (
-        <div class="product-card" key={product.id}>
+        <div className="product-card" key={_id}>
             <div>
-                <div class="card-top-half">
-                    <i class="fas fa-regular fa-heart product-fav-icn fav-active"></i>
+                <div className="card-top-half">
+                    <i
+                        className={
+                            checkWishlist
+                                ? 'fas fa-regular fa-heart product-fav-icn fav-active'
+                                : 'far fa-regular fa-heart product-fav-icn fav-active'
+                        }
+                        onClick={
+                            !checkWishlist
+                                ? () => addToWishlistHandler(product)
+                                : () => removeFromWishlistHandler(_id)
+                        }
+                    ></i>
                     <div className="product-asset-container">
-                        <img src={product.asset} alt="product body asset" class="product-body-asset" />
+                        <img src={asset} alt="product body asset" className="product-body-asset" />
                     </div>
-                    
-                    <div class="card-top-half-text">
-                        <p class="product-item-title">{product.title}</p>
-                        <p class="product-item-price">Rs {product.price}</p>
+
+                    <div className="card-top-half-text">
+                        <p className="product-item-title">{title}</p>
+                        <p className="product-item-price">
+                            Rs {(price - (price * discount) / 100).toFixed(2)}{' '}
+                            <strike>{price}</strike>{' '}
+                        </p>
+                        <p className="product-item-price">{discount}% off!</p>
                     </div>
                 </div>
-                <button class="product-item-cta">View Product</button>
+                <button
+                    className="product-item-cta"
+                    onClick={
+                        auth
+                            ? checkCart
+                                ? () => updateHandler(product, 'increment')
+                                : () => addToCartHandler(product)
```

Comment:
If an item already exists in the cart, you have to show 'Go to Cart' and navigate to cart on click of the button